File: TodosApp-Vite/src/services/TodosService.ts
import { fetch } from "@/libs/Fetch";

class TodosService {
  public getTodos() {
    return fetch
      .get("/get-todos")
      .then((response) => {
        return response.data;
      })
      .catch((error) => {
        console.error(error);
      });
  }

  public getTodo(id: string) {
    fetch
      .get("/get-todo", { params: { id: id } })
      .then(() => {})
      .catch(() => {});
  }

  public createTodo(title: string, description: string, callback?: () => void) {
    fetch
      .post(
        "/add-todo",
        {},
        {
          params: {
            title: title,
            description: description,
          },
        }
      )
      .then(() => {
        callback && callback();
      })
      .catch((error) => {
        console.error(error);
      });
  }

  public removeTodo(id: string, callback?: () => void) {
    fetch
      .delete("/delete-todo", {
        params: {
          id: id,
        },
      })
      .then(() => {
        callback && callback();
      })
      .catch((error) => {
        console.error(error);
      });
  }
}

export default new TodosService();

File: TodosApp-Springboot/src/main/java/me/teelyjc/app/services/TodosService.java
package me.teelyjc.app.services;

import java.util.ArrayList;

import me.teelyjc.app.domains.Todo;
import me.teelyjc.app.repositories.TodosRepository;

public class TodosService {
    private TodosRepository todosRepository = new TodosRepository();

    public void createTodo(String title, String description) {
        this.todosRepository.createTodo(title, description);
    }

    public void removeTodo(String id) {
        this.todosRepository.removeTodo(id);
    }

    public ArrayList<Todo> getTodos() {
        return this.todosRepository.getTodos();
    }

    public ArrayList<Todo> getTodo(String id) {
        return this.todosRepository.getTodo(id);
    }
}
